### PR TITLE
docs: Add example on query parameters in route handlers

### PR DESCRIPTION
**Pull request by joulev**
### Why?

There is already an example of handling query parameters (`searchParams`) in route handler, but it's quite hidden (in [Opting out of caching](https://nextjs.org/docs/app/building-your-application/routing/route-handlers#opting-out-of-caching)). It made it hard for many developers to figure out how to do that: [there](https://discord.com/channels/752553802359505017/1127781032242974790) [are](https://discord.com/channels/752553802359505017/752647196419031042/1143181251856044042) [several](https://discord.com/channels/752553802359505017/752647196419031042/1146007247630106704) [people](https://discord.com/channels/752553802359505017/752647196419031042/1154623267164917861) who couldn't find that section in the documentation. (Linked are messages in the Next.js Discord server.)

Therefore this PR adds an example to the route handler documentation on how to do that, so developers can take reference more easily.

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: eab4dd30978075374114f5557c1e4638300599b0

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: eab4dd30978075374114f5557c1e4638300599b0

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: eab4dd30978075374114f5557c1e4638300599b0

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

File: docs/02-app/01-building-your-application/01-routing/10-route-handlers.mdx [hunk 1]
```diff
@@ -386,6 +386,28 @@ export async function GET(request, { params }) {
 | `app/items/[slug]/route.js` | `/items/b`  | `{ slug: 'b' }` |
 | `app/items/[slug]/route.js` | `/items/c`  | `{ slug: 'c' }` |
 
+### URL Query Parameters
+
+The request object passed to the route handler is a `NextRequest` instance, which has [some additional convenient methods](/docs/app/api-reference/functions/next-request#nexturl), including for handling query parameters.
```
Comment (leerob) [inline docs/02-app/01-building-your-application/01-routing/10-route-handlers.mdx:391]:
```suggestion
The request object passed to the Route Handler is a `NextRequest` instance, which has [some additional convenience methods](/docs/app/api-reference/functions/next-request#nexturl), including more easily handling query parameters.
```

File: docs/02-app/01-building-your-application/01-routing/10-route-handlers.mdx [hunk 2]
```diff
@@ -386,6 +386,28 @@ export async function GET(request, { params }) {
...
+
+```ts filename="app/test/route.ts" switcher
```
Comment (leerob) [inline docs/02-app/01-building-your-application/01-routing/10-route-handlers.mdx:393]: Should we do a more realistic route name? Maybe `/api/name`

**Review by leerob — APPROVED** [review]

**Pull request merged**